### PR TITLE
Cleanup Windows backend code a bit

### PR DESCRIPTION
**Pull request by DJm00n**
Cleanup Windows backend code a bit before adding HID device connection callback code.

### CODE REVIEW

**Comment by DJm00n** [comment]
@Youw could you please review this?

File: windows/hid.c [hunk 1]
```diff
@@ -306,14 +309,116 @@ int HID_API_EXPORT hid_exit(void)
 	return 0;
 }
 
+static struct hid_device_info* hid_get_device_info(const char* path)
```
Comment (Youw) [inline windows/hid.c:312]: please keep _consistency_ across the code base, spaces _before_ the asterisk:
```suggestion
static struct hid_device_info *hid_get_device_info(const char *path)
```
this applias to everything below too

File: windows/hid.c [hunk 2]
```diff
@@ -398,117 +507,43 @@ struct hid_device_info HID_API_EXPORT * HID_API_CALL hid_enumerate(unsigned shor
...
-
-			#define WSTR_LEN 512
-			const char *str;
-			struct hid_device_info *tmp;
-			PHIDP_PREPARSED_DATA pp_data = NULL;
-			HIDP_CAPS caps;
-			NTSTATUS nt_res;
-			wchar_t wstr[WSTR_LEN]; /* TODO: Determine Size */
-			size_t len;
+			(product_id == 0x0 || attrib.ProductID == product_id)) {
```
Comment (Youw) [inline windows/hid.c:530]: revert this line

File: windows/hid.c [hunk 3]
```diff
@@ -551,7 +585,7 @@ HID_API_EXPORT hid_device * HID_API_CALL hid_open(unsigned short vendor_id, unsi
 	cur_dev = devs;
 	while (cur_dev) {
 		if (cur_dev->vendor_id == vendor_id &&
-		    cur_dev->product_id == product_id) {
+			cur_dev->product_id == product_id) {
```
Comment (Youw) [inline windows/hid.c:588]: revert this line

**Review by Youw — COMMENTED** [review]
So now we open the device/handle 2 times during the enumeration.
I'd strongly suggest to cache/reuse the same handle, even if some of the sub-routines are in a separate function.

**Review by Youw — APPROVED** [review]

**Comment by DJm00n** [comment]
@Youw can you merge this?

**Pull request merged**